Add Alert Wrapper

marvinbuff opened the pull request:
Hi, thanks for the awesome library. I know everything is still WIP, so I thought I would make some contributions while using it. 

I currently need to use the Alert component which does not yet have any wrapper. So I  just created one. 

Let me know if you have any feedback or requested changes. Especially in regards to the dependency to `material-ui/lab`. Not sure if you don't want to depend on the experimental library or if you just didn't get to create wrappers for it yet.

cfnz commented:
Hi there, thanks for that.

I mainly have not included the lab stuff more for the experimental nature of it, but I don't mind adding it if it is useful. Will be interesting to see how the Kotlin DCE deals with it.

I will add a couple of comments in the code review too :-)

cfnz commented on .gitignore:
```diff
@@ -106,3 +106,6 @@ typings/
 /build/
+
+# Intellij
+.idea
```
:+1:

cfnz commented on muirwik-components/src/main/kotlin/com/ccfraser/muirwik/components/Snackbar.kt:
```diff
@@ -64,7 +64,7 @@ var MSnackbarProps.transitionDuration by TransitionDurationDelegate()
 
 fun RBuilder.mSnackbar(
-        message: ReactElement,
+        message: ReactElement?,
```
I am wondering if it would be better to leave this non-nullable and add a new function without the message parameter (or swap the order of the params, but not sure if many people have used mSnackbar already)... mainly so we don't have to use a null when calling it...

So fun Rbuilder.mSnackbar (open: Boolean... etc)

cfnz commented on muirwik-testapp/src/main/kotlin/com/ccfraser/muirwik/testapp/TestAlert.kt:
```diff
@@ -0,0 +1,49 @@
+        css(margin)
+      }
+
+      mSnackbar(null, open = open) {
```
If we go with an extra mSnackbar function without the message as per my comment on the component, this would then become:

`mSnackbar(open)`

marvinbuff commented:
Implemented the changes as requested. I also added some comments which hopefully help to understand the difference between the builders.

As a side note: I will use this library extensively for some time. Are you interested in more pull requests? For instance I would have some changes to the table to make it more generic, and I would like to have some utility changes to work on it. Do you generally welcome such changes?

cfnz commented:
Yes, happy to receive pull requests, I might be a bit slow turning them around, but I do hope to do so. 

I added the AlertTitle to your pull request and made a few tweaks (hope you don't mind :-)).

But yes, the more people improving the better :-).
